frontend/awx/resources/hosts/Hosts.cy.tsx

```diff
@@ -9,15 +9,32 @@ describe('Hosts.cy.ts', () => {
   const hosts = 'hosts';
   const inventory_hosts = 'inventory_hosts';
   const smart_inventory_hosts = 'smart_inventory_hosts';
+  const constructed_inventory_hosts = 'constructed_inventory_hosts';
 
-  const types = [hosts, inventory_hosts, smart_inventory_hosts];
+  const types = [hosts, inventory_hosts, smart_inventory_hosts, constructed_inventory_hosts];
 
   types.forEach((type) => {
     const typeDesc = ` (${type})`;
     const component = type === hosts ? <Hosts /> : <InventoryHosts />;
     const path = type === hosts ? '/hosts' : '/inventories/:inventory_type/:id/hosts';
 
-    const inventory_type = type === smart_inventory_hosts ? 'smart_inventory' : 'inventory';
+    let inventory_type = '';
```

Comment:
I understand we are trying to reuse code that would otherwise be duplicated for each type of inventory.  However, I tend to think that in this case duplicating the code....or making it less complex is a good thing.  It helps us quickly identify any test failures that may get caught in the future.

Reply:
I do see advantage of your approach, but having so much duplicated code will be bad too. What about this compromise - create functions for duplicated its and call the issue of those its and get rid of the foreach? So it will be more linear but with reused functions.